### DISCUSSION TITLE
resolver not reported correctly

### DISCUSSION THREAD

**Discussion by ghost** [DNSCrypt/dnscrypt-proxy#2389] [Potential issues]
When using the `-resolve` option DNSCrypt-Proxy always shows an external encrypted resolver even if the result is actually produced locally because of forwarding, blocklist, etc.
Isn't that misleading and should be changed?

**Comment by lifenjoiner** [comment]
Maybe a better name is needed. Not just "Resolver".

I was confused on this sub-command too. Finally, I realized:
1. It is not an alternative to `dig`.
2. It is a proof of working, for:
2.1 There is an upstream server working.
2.2 The upstream server can resolve the domain, perfectly normally, in special case not.
2.3 The displayed resolver doesn't correspond to the following answers.

Capture the packets with `Wireshark`/`tcpdump`, you will see how it works.
And, there is a "secret mission".
